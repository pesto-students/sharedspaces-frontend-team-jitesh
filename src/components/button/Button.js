import React from 'react'
import './button.scss'
import Loader from '../loader/Loader'

export default function Button({ loading, children, type, buttonType, className, onSubmit, onClick }) {
    return (
        <button
            data-testid="button"
            type={type}
            onSubmit={onSubmit}
            onClick={onClick}
            className={`
            btn-wrapper
        ${buttonType === "primary" && "bg-red-500 text-white border-2 border-red-500 hover:bg-red-500 hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white"}
        ${buttonType === "primary-outline" && "bg-white text-red-500 border-red-500 border-2 hover:bg-red-500 hover:text-white hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white"}
        ${buttonType === "dark" && "bg-gray-800 text-white border-2 border-gray-800 hover:bg-gray-800 hover:shadow-sm active:bg-gray-800 active:shadow-sm active:text-white"}
        ${buttonType === "dark-outline" && "bg-white text-gray-800 border-gray-800 border-2 hover:bg-gray-800 hover:text-white hover:shadow-sm active:bg-red-800 active:shadow-sm active:text-white"}
        ${className}
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer`}>
            {loading ? <Loader width={"w-5"} /> : children}
        </button>
    )
}
