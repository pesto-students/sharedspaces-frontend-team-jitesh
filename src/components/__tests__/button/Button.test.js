import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import Button from '../../button/Button'

afterEach(() => {
    cleanup()
})

test('test matches snapshot', () => {
    const component = renderer.create(<Button />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})


test('should render button component', () => {
    render(<Button />)

    const buttonElement = screen.getByTestId("button")

    expect(buttonElement).toBeInTheDocument();
})