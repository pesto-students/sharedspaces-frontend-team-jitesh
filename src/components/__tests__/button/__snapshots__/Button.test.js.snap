// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<button
  className="
            btn-wrapper
        false
        false
        false
        false
        undefined
        px-6 py-1.5 font-medium rounded shadow-md transition duration-150 ease-in-out  whitespace-nowrap cursor-pointer"
  data-testid="button"
/>
`;
